### PR TITLE
add local action readme documentation

### PR DESCRIPTION
**Pull request by mattdavis0351**
This pull adds documentation to the local action.

### CODE REVIEW

File: .github/actions/check-secrets-action/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Check Secrets Action
```
Comment (lucie-docs) [inline .github/actions/check-secrets-action/README.md:1]:
```suggestion
# Check-secrets action
```

File: .github/actions/check-secrets-action/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+This local action runs whenever a change to the `.github/workflows/use-secrets.yml` is pushed to your repository.
+
+There are two inputs necessary for this action to run and provide a grade for this exercise:
+
+| Input           | Description                                                                                                                                                                                                                             | Required parameter |
```
Comment (lucie-docs) [inline .github/actions/check-secrets-action/README.md]:
```suggestion
| Input           | Description                                                                                                                                                                                                                             | Required |
```

File: .github/actions/check-secrets-action/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+| --------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | :----------------: |
+| secrets-context | This a JSON formatted Object containing the [secrets context](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#contexts) for the repository from which the `use-secrets.yml` was triggered |         ✅         |
```
Comment (lucie-docs) [inline .github/actions/check-secrets-action/README.md]:
```suggestion
| `secrets-context` | JSON-formatted object containing the [secrets context](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#contexts) for the repository from which the `use-secrets.yml` workflow was triggered |         Yes         |
```

Applying the Microsoft accessibility guidelines regarding the use of emojis.

File: .github/actions/check-secrets-action/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+| your-secret     | The value of this input should be a Personal Access Token containing the repo scope. It should be passed in using repository secrets. **Do not** place the raw value of the Personal Access Token into the `use-secrets.yml` workflow!  |         ✅         |
```
Comment (lucie-docs) [inline .github/actions/check-secrets-action/README.md]:
```suggestion
| `your-secret`     | The value of this input should be a personal access token (PAT) with a repo scope. This PAT should be passed in the form of a repository secret in the `use-secrets.yml` workflow . **Do not** place the raw value of the PAT into the `use-secrets.yml` workflow!  |         Yes         |
```

File: .github/actions/check-secrets-action/README.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+## What checks does this action conduct?
+
+- Counts the number of secrets the repository has to ensure the leaner added a secret.
+- Attempts to perform an HTTP request using Octokit and the provided secret to validate that the secret contains a valid PAT.
```
Comment (hectorsector) [inline .github/actions/check-secrets-action/README.md:15]: @jonlipsky we discussed this yesterday, turns out we're already doing it! 👀

**Review by hectorsector — APPROVED** [review]
Thanks @mattdavis0351! I'm loving the clarity that's coming with this.

File: .github/actions/check-secrets-action/README.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+| --------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | :----------------: |
+| secrets-context | This a JSON formatted Object containing the [secrets context](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#contexts) for the repository from which the `use-secrets.yml` was triggered |         ✅         |
+| your-secret     | The value of this input should be a Personal Access Token containing the repo scope. It should be passed in using repository secrets. **Do not** place the raw value of the Personal Access Token into the `use-secrets.yml` workflow!  |         ✅         |
+
+## What checks does this action conduct?
+
+- Counts the number of secrets the repository has to ensure the leaner added a secret.
```
Comment (lucie-docs) [inline .github/actions/check-secrets-action/README.md]:
```suggestion
- Counts the number of secrets the repository has to ensure the learner added a secret.
```

File: .github/actions/check-secrets-action/README.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+- Attempts to perform an HTTP request using Octokit and the provided secret to validate that the secret contains a valid PAT.
+- Reports errors and failures in the actions runner of the `use-secrets.yml` workflow
+- Sends certain errors and success messages to the `grading.yml` workflow
```
Comment (lucie-docs) [inline .github/actions/check-secrets-action/README.md]:
```suggestion
- Sends certain error and success messages to the `grading.yml` workflow
```

File: .github/actions/check-secrets-action/README.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+## What happens with your Personal Access Token?
```
Comment (lucie-docs) [inline .github/actions/check-secrets-action/README.md:19]:
```suggestion
## What happens with your personal access token?
```
Aligning casing to casing in docs and UI.

File: .github/actions/check-secrets-action/README.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
+A personal Access Token is needed to allow the `use-secrets.yml` workflow to communicate to the `grading.yml` workflow using a [`repository_dispatch`](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#repository_dispatch) event.
```
Comment (lucie-docs) [inline .github/actions/check-secrets-action/README.md:21]:
```suggestion
A personal access token (PAT) is needed to allow the `use-secrets.yml` workflow to communicate to the `grading.yml` workflow using a [`repository_dispatch`](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#repository_dispatch) event.
```

File: .github/actions/check-secrets-action/README.md [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+
+There is a built in feature of GitHub Actions that prevents one workflow from triggering another, even through the use of a `repository_dispatch` event, if the authentication for the workflow is provided by the `GITHUB_TOKEN`.
```
Comment (lucie-docs) [inline .github/actions/check-secrets-action/README.md:23]:
```suggestion
A built-in feature of GitHub Actions prevents one workflow from triggering another, even through the use of a `repository_dispatch` event, if the authentication for the workflow is provided by the `GITHUB_TOKEN`.
```

File: .github/actions/check-secrets-action/README.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+
+However, if the first workflow is supplied a [properly scoped](https://docs.github.com/en/rest/reference/repos#create-a-repository-dispatch-event) Personal Access Token, GitHub Actions will allow this workflow to trigger other workflow runs.
```
Comment (lucie-docs) [inline .github/actions/check-secrets-action/README.md:25]:
```suggestion
However, if the first workflow is supplied a [properly scoped](https://docs.github.com/en/rest/reference/repos#create-a-repository-dispatch-event) PAT, GitHub Actions will allow this workflow to trigger other workflow runs.
```

File: .github/actions/check-secrets-action/README.md [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+
+In the `use-secrets-action` we use [Octokit](https://github.com/octokit) to create the repository dispatch using your PAT as the authentication that allows for one workflow to trigger another. You can read more about the [Octokit method](https://octokit.github.io/rest.js/v18#repos-create-dispatch-event) being used to make this HTTP request if you'd like to learn more.
```
Comment (lucie-docs) [inline .github/actions/check-secrets-action/README.md:27]:
```suggestion
In the `check-secrets-action` we use [Octokit](https://github.com/octokit) to create the repository dispatch using your PAT as the authentication that allows for one workflow to trigger another. You can read more about the [Octokit method](https://octokit.github.io/rest.js/v18#repos-create-dispatch-event) being used to make this HTTP request if you'd like to learn more.
```

File: .github/actions/check-secrets-action/README.md [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+
+## Summary
+
+- The PAT you create **stays in your repo**
+- The `use-secrets-action` never exposes the value of the secret you pass into the action
```
Comment (lucie-docs) [inline .github/actions/check-secrets-action/README.md]:
```suggestion
- The `check-secrets-action` never exposes the value of the secret you pass into the action.
```

File: .github/actions/check-secrets-action/README.md [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+
+There is a built in feature of GitHub Actions that prevents one workflow from triggering another, even through the use of a `repository_dispatch` event, if the authentication for the workflow is provided by the `GITHUB_TOKEN`.
+
+However, if the first workflow is supplied a [properly scoped](https://docs.github.com/en/rest/reference/repos#create-a-repository-dispatch-event) Personal Access Token, GitHub Actions will allow this workflow to trigger other workflow runs.
+
+In the `use-secrets-action` we use [Octokit](https://github.com/octokit) to create the repository dispatch using your PAT as the authentication that allows for one workflow to trigger another. You can read more about the [Octokit method](https://octokit.github.io/rest.js/v18#repos-create-dispatch-event) being used to make this HTTP request if you'd like to learn more.
+
+## Summary
+
+- The PAT you create **stays in your repo**
```
Comment (lucie-docs) [inline .github/actions/check-secrets-action/README.md]:
```suggestion
- The PAT you create **stays in your repo**.
```

File: .github/actions/check-secrets-action/README.md [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+- The `use-secrets-action` never exposes the value of the secret you pass into the action
+- An HTTP request is made using Octokit with the supplied PAT for authentication
```
Comment (lucie-docs) [inline .github/actions/check-secrets-action/README.md]:
```suggestion
- An HTTP request is made using Octokit with the supplied PAT for authentication.
```

File: .github/actions/check-secrets-action/README.md [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+- A report is generated and displayed in the actions runner
```
Comment (lucie-docs) [inline .github/actions/check-secrets-action/README.md]:
```suggestion
- A report is generated and displayed in the actions runner.
```

File: .github/actions/check-secrets-action/README.md [hunk 17]
```diff
@@ -0,0 +1,34 @@
+# Check Secrets Action
+
+This local action runs whenever a change to the `.github/workflows/use-secrets.yml` is pushed to your repository.
```
Comment (lucie-docs) [inline .github/actions/check-secrets-action/README.md]:
```suggestion
This local action runs whenever a change to the `.github/workflows/use-secrets.yml` workflow is pushed to your repository.
```

**Review by lucie-docs — CHANGES_REQUESTED** [review]
Hi @mattdavis0351, suggested a few changes (mostly cosmetic) 🙂

File: .github/actions/check-secrets-action/README.md (hunk 3 above)
Comment (mattdavis0351) [reply .github/actions/check-secrets-action/README.md]: @lucie-docs Object with a capital O is grammatically correct.  In JavaScript an Object is a noun and is almost always referred to with a capital O in any and all documentation you will come across.

Here is a quick look at [the official spec for JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object)

Comment (lucie-docs) [reply .github/actions/check-secrets-action/README.md]: When it's a class yeah. But I believe in this context you're using it as a noun or am I missing something? 🙂 
This is from the link you sent: "Nearly all objects in JavaScript are instances of `Object`; (...)". First occurrence of object is a noun and not capitalized. Second one refers to the class and is capitalized and between ``. I recommend using the same convention so either it's a class and we write it as `Object` or it's a noun and in this case we write it as object.

**Comment by hectorsector** [comment]
Thanks for the feedback and commits, @lucie-docs and @mattdavis0351! Is this ready to merge?

File: .github/actions/check-secrets-action/README.md (hunk 3 above)
Comment (mattdavis0351) [reply .github/actions/check-secrets-action/README.md]: @lucie-docs fair point.  Small o works for me.  If learners take up issue with it I'm assuming someone will open a PR to fix it.  We get a lot of that with LL courses.

**Comment by mattdavis0351** [comment]
@hectorsector it's ready from my end .  Merging now that all @lucie-docs changes have been implemented.

**Pull request merged**